### PR TITLE
add amorized inference example

### PR DESCRIPTION
**Pull request by kiante-fernandez**
This is the first of two examples I want to include for using simulation-based inference with models in SSM.jl.

Here, I am using the package NeuralEstimators.jl. Feedback is welcome at this point.

https://github.com/msainsburydale/NeuralEstimators

Next, I am working on building likelihood approximation networks in Flux based on methods from Fengler, and I will have it up and running to sample with Turing. That will be the next example.

For now, I tried to use the LCA, as similar examples have shown online (see https://bayesflow.org/stable-legacy/_examples/LCA_Model_Posterior_Estimation.html).

I was able to get this running local but I think we have to env coflict we have to resolve to get NeuralEstimators working on the docs

### CODE REVIEW

**Comment by itsdfish** [comment]
Thanks! I only gave it a cursory look, but this is nice from what I can tell. 

> I was able to get this running local but I think we have to env coflict we have to resolve to get NeuralEstimators working on the docs

I think that is OK. I no longer auto-generate computationally intensive docs with the `@example` block to cut down on run time. 

By the way, I have been corresponding with the developer of NeuralEstimators.jl to get help with a few technical details. One thing I have planned is to add an example of Bayesian parameter estimation of the LNR using NormalisingFlow and comparing it to Turing (for validation). I want to let you know so we do not duplicate any work. Thanks again!

**Comment by kiante-fernandez** [comment]
> Thanks! I only gave it a cursory look, but this is nice from what I can tell.
> 
> > I was able to get this running local but I think we have to env coflict we have to resolve to get NeuralEstimators working on the docs
> 
> I think that is OK. I no longer auto-generate computationally intensive docs with the `@example` block to cut down on run time.
> 
> By the way, I have been corresponding with the developer of NeuralEstimators.jl to get help with a few technical details. One thing I have planned is to add an example of Bayesian parameter estimation of the LNR using NormalisingFlow and comparing it to Turing (for validation). I want to let you know so we do not duplicate any work. Thanks again!

Okay, I will add `@example`. Also, nice! I think both cases are good examples, since in one we can map it to the "ground truth" by comparing with Turing.

I want to follow up with the team, as I think some additional text on considering simulation-based calibration could be instructive. Do you know if they are working on that? I was considering opening an issue.

**Comment by itsdfish** [comment]
> Okay, I will add `@example`.

Just to clarify, please to not use `@example` for long running documentation such as this.  One thing that would be helpful is if you included a full, copy and paste version of the code in the following:


```julia
#```@raw html
#<details>
#<summary><b>Show Details </b></summary>
#```
#```julia
## your code here
#```
#```@raw html
#</details>
#```
```
Sorry for the `#`. It would not display at all otherwise. Anyways, this will allow users to copy and paste a full version of the code rather than multiple pieces. It also hides/reveals the code so it does not create visual clutter. I also recommend it for testing the examples locally, since `@example` is too slow to run each time we test the docs. 



> I want to follow up with the team, as I think some additional text on considering simulation-based calibration could be instructive. Do you know if they are working on that? I was considering opening an issue.

Not that I know of. It might be worth looking into.

File: docs/src/neuralestimators_amorized.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+using Plots
+
+Random.seed!(123)
+```
+
+## Define Parameter Bounds
+
+```julia
+# Define parameter bounds for the LCA model
```
Comment (itsdfish) [inline docs/src/neuralestimators_amorized.md:28]: Given that  being greater than zero is the only hard boundary, I  wonder whether `Dense(w, 1, softplus)` be a simpler way to constrain the parameter estimates? Does that not work?

If not, I recommend either passing these are arguments to `create_neural_estimator`, or hard coding them into `create_neural_estimator`.

File: docs/src/neuralestimators_amorized.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+const ν_min, ν_max = 0.1, 4.0      # Drift rates
+const α_min, α_max = 0.5, 3.5      # Threshold
+const β_min, β_max = 0.0, 0.5      # Lateral inhibition
+const λ_min, λ_max = 0.0, 0.5      # Leak rate
+const τ_min, τ_max = 0.1, 0.5      # Non-decision time
+```
+
+## Define Parameter Sampling
+
+Unlike traditional Bayesian approaches, simulation based inference methods require us to define a prior sampling function to generate training data. We will use this function to sample a range of parameters for training:
```
Comment (itsdfish) [inline docs/src/neuralestimators_amorized.md:38]: I was a bit confused about this statement. Do you include MCMC in your definition of traditional? MCMC  does sample values from a prior distribution. Do you mean analytic approaches?

**Comment by kiante-fernandez** [comment]
> Sorry for the `#`. It would not display at all otherwise. Anyways, this will allow users to copy and paste a full version of the code rather than multiple pieces. It also hides/reveals the code so it does not create visual clutter. I also recommend it for testing the examples locally, since `@example` is too slow to run each time we test the docs.
> 
> >

I see understood! I will modify.

File: docs/src/neuralestimators_amorized.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+```julia
+# Function to sample parameters from priors
+function sample(K::Integer)
+    ν1 = rand(Gamma(2, 1/1.2), K)  # Drift rate 1
+    ν2 = rand(Gamma(2, 1/1.2), K)  # Drift rate 2
+    α = rand(Gamma(10, 1/6), K)    # Threshold
+    β = rand(Beta(1, 5), K)        # Lateral inhibition
+    λ = rand(Beta(1, 5), K)        # Leak rate
```
Comment (itsdfish) [inline docs/src/neuralestimators_amorized.md:47]: Its slightly ugly, but one way to generate `Vector{Float32}` for efficiency is to use `rand(Gamma(2f0, 1/1.2f0), K)`

Comment (itsdfish) [reply docs/src/neuralestimators_amorized.md:47]: You could simplify with `rand(Gamma(2, 1/1.2f0), K)`. Since the first argument is an integer, it will promote the `Float32`

File: docs/src/neuralestimators_amorized.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+        σ = 1.0        # Fixed diffusion noise
+        
+        # Create LCA model
+        model = LCA(; ν, α, β, λ, τ, σ)
+        
+        # Generate choices and reaction times
+        choices, rts = rand(model, n_trials_per_param)
+        
+        # Return as a transpose matrix where each column is a trial 
+        return Float32.([choices rts]')
```
Comment (itsdfish) [inline docs/src/neuralestimators_amorized.md:77]: If you use `f0` to use `Float32` in your `sample`, I think you can get rid of this conversion. If that does not work, open an issue, and I can try to make it more generic.

File: docs/src/neuralestimators_amorized.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+recovered_params = NeuralEstimators.estimate(trained_estimator, [observed_data])
+
+# Compare true and recovered parameters
+println("True parameters: ", [ν[1], ν[2], α, β, λ, τ])
+println("Recovered parameters: ", recovered_params)
+```
+
+## Notes on Performance
+
+Neural estimators are particularly effective for models with computationally intractable likelihoods like the LCA model. However, certain parameters (particularly β and λ) can be difficult to recover, even with advanced neural network architectures. This is a  property of the LCA model rather than a limitation of the estimation technique.
```
Comment (itsdfish) [inline docs/src/neuralestimators_amorized.md:248]: Good point to make!

File: docs/src/neuralestimators_amorized.md [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+
+A key advantage of neural estimation is the ability to quickly conduct inference after training. For example, we can visualize the recovery of parameters:
+
+```julia
+# Extract data from assessment
+df = assessment.df
+
+# Create recovery plots for each parameter
+params = unique(df.parameter)
+p_plots = []
```
Comment (itsdfish) [inline docs/src/neuralestimators_amorized.md:194]: NeuralEstimators includes similar functionality with `AlgebraOfGraphics.plot(assessment)`. Not that you need to use it but it could be convienent. If decide not to use it, can you make a brief note that it is available?

File: docs/src/neuralestimators_amorized.md (hunk 2 above)
Comment (kiante-fernandez) [reply docs/src/neuralestimators_amorized.md:38]: You're right that MCMC samples from the prior. What I meant is that in simulation-based inference, we explicitly generate synthetic datasets by sampling parameters from a prior and simulating data under those parameters—that is, we use the prior to generate training data for a model (e.g., a neural network). In contrast, traditional Bayesian methods like MCMC don’t use prior samples to train a model; they use them as part of the inference process. I was distinguishing this method from approaches where the prior isn't used to generate a large training set upfront. I appreciate the opportunity to clarify.

File: docs/src/neuralestimators_amorized.md (hunk 1 above)
Comment (kiante-fernandez) [reply docs/src/neuralestimators_amorized.md:28]: The sigmoid scaling approach seems appropriate for maintaining these specific bounds and providing a little more control. I will modify it to move code into the estimator.

**Review by itsdfish — CHANGES_REQUESTED** [review]
Kiante, there a few minor revisions requested below. Once we sort those out, I will merge.

File: docs/src/neuralestimators_amorized.md [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+    p = scatter(
+        param_data.truth, 
+        param_data.estimate,
+        xlabel="Ground Truth",
+        ylabel="Estimated",
+        title=param,
+        legend=false
+    )
+    plot!(p, [minimum(param_data.truth), maximum(param_data.truth)], 
+          [minimum(param_data.truth), maximum(param_data.truth)],
```
[the remark this answers is not in the document]
Comment (kiante-fernandez) [reply docs/src/neuralestimators_amorized.md:209]: I have updated the training and achieved much better recovery for the example. A new figure has been added.

**Comment by itsdfish** [comment]
@kiante-fernandez, thank you for the revisions. This looks nice. I will go ahead and merge.


The unit tests for Turing is not passing with 0.37, even though it passed [before](https://github.com/itsdfish/SequentialSamplingModels.jl/actions/runs/14292360104/job/40055228396). I'm looking into a solution.
